src/hooks/useCalculator.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from 'react';
+import {
+    caculateResultInput,
+    isValidOperationInput,
+    isValidResultInput,
+} from '../utils';
+
+export const useCalculator = () => {
+    const [digitInput, setDigitInput] = useState<string>('');
+    const [operationInput, setOperationInput] = useState<string>('');
+    const [resultInput, setResultInput] = useState<string>('0');
+    const [operatorPrevNumber, setOperatorPrevNumber] = useState<string>('');
+
+
+    // 1.  들어오는 DigitInput에따라 resultInput 결과를 처리한다 .
+    const handleDigitInput = (e: React.MouseEvent<HTMLButtonElement>) => {
+        
+        // 1.1 숫자 유효성 체크
+        const currentDigitNumber = parseInt(
+            digitInput + e.currentTarget.value
+        ).toString(); //000입력되는거 방지
+
+        if (currentDigitNumber.length > 3) {
+            alert('숫자는 최대 3자리까지 입력가능합니다.');
+            return;
+        }
+
+        // 1.2 OperationInput 존재에 따라 ResultInput을 보여줄지 처리
+        if (!operationInput.length) { //operation이 존재 안하면 현재 숫자만 보여주면됨        
+            setResultInput(currentDigitNumber);
+            // Operator 이전 숫자를 저장하기 위해 사용
+            setOperatorPrevNumber(currentDigitNumber);
+        } else {
+            // 2. opeartion 존재할시 
+            //operation 이전에 있는 숫자 (operatorPrevNumber) + 현재 currentDigitNumber = resultINput
+            const currentResultInput =
+                operatorPrevNumber + operationInput + currentDigitNumber;
+            setResultInput(currentResultInput);
+        }
+        // 현재입력된 DigitInput갱신
+        setDigitInput(currentDigitNumber);
+    };
+
+    // 2. OperationInput이 들어올경우처리하는 함수
+    //  OperationInput이 처음 생길때와 존재하지 않을때에 따라서 처리하는 방식을 나눔
+    const handleOperationInput = (e: React.MouseEvent<HTMLButtonElement>) => {
+        const currentOperation = e.currentTarget.value; //현재 operation
+        if (operationInput.length) {  // operationInput이 존재할떄
+            if (isValidOperationInput(currentOperation)) return; // = 이 아니면 operation못들가게
```

Comment:
함수를 사용하지 않으면 주석도 필요없을 것 같아요
```suggestion
            if (currentOperation === "=") return;
```